src/analysis/normed_space/basic.lean

```diff
@@ -1947,6 +1947,118 @@ summable_of_nnnorm_bounded _ hf (assume i, le_refl _)
 
 end summable
 
+section cauchy_product
+
+/-! ## Multipliying two infinite sums in a normed ring
+
+In this section, we prove various results about `(∑' x : ι, f x) * (∑' y : ι', g y)` in a normed
+ring. There are similar results proven in `topology/algebra/infinite_sum` (e.g `tsum_mul_tsum`),
+but in a normed ring we get summability results which aren't true in general.
+
+We first establish results about arbitrary index types, `β` and `γ`, and then we specialize to
+`β = γ = ℕ` to prove the Cauchy product formula (see `tsum_mul_tsum_nat_of_summable_norm`).
+
+### Arbitrary index types
+-/
+
+variables {ι' : Type*} [normed_ring α]
+
+open_locale classical
+
+lemma real.summable_mul_of_summable_of_nonneg {f : ι → ℝ} {g : ι' → ℝ}
```

Comment:
Is there any point in putting that in the `real` namespace? Here I'm thinking in terms of dot notation. For instance, you could rename that lemma `summable.mul_of_nonneg` and then the next proof would be 
```
summable_of_nonneg_of_le (λ x, norm_nonneg (f x.1 * g x.2)) (λ x, norm_mul_le (f x.1) (g x.2))
  (hf.mul_of_nonneg hg (λ x, norm_nonneg $ f x) (λ x, norm_nonneg $ g x))
```

Reply:
Yes that's better indeed. I am too obsessed with the use of namespaces as a mean of classification.

Reply:
About the next proof, I noticed that your version takes a significant times to be elaborated (multiple seconds on my computer), but replacing `hf.mul_of_nonneg hg`   by `@summable.mul_of_nonneg _ _ _ _ hf hg` makes it almost instant. What is the convention for such situations, readability or performance ?